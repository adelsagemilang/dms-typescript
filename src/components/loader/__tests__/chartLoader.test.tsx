import React from 'react';
import { shallow } from 'enzyme';
import XYChart from '../chart';

describe('Chart Loader tests', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<XYChart height={400} width={400} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly while props changed', () => {
    const wrapper = shallow(<XYChart height={400} width={400} />);
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({
      height: 300,
      width: 300,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
