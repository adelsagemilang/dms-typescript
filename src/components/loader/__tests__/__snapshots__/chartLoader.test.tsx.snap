// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Loader tests renders correctly 1`] = `
<ContentLoader
  height={400}
  speed={1}
  viewBox="0 0 400 400"
  width="100%"
>
  <rect
    height="1"
    rx="0"
    ry="0"
    width={400}
    x="20"
    y="175"
  />
  <rect
    height="100"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="40"
    y="75"
  />
  <rect
    height="50"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="80"
    y="125"
  />
  <rect
    height="70"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="120"
    y="105"
  />
  <rect
    height="140"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="160"
    y="35"
  />
  <rect
    height="120"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="200"
    y="55"
  />
  <rect
    height="160"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="240"
    y="15"
  />
  <rect
    height="40"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="280"
    y="135"
  />
  <rect
    height="90"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="320"
    y="85"
  />
</ContentLoader>
`;

exports[`Chart Loader tests renders correctly while props changed 1`] = `
<ContentLoader
  height={400}
  speed={1}
  viewBox="0 0 400 400"
  width="100%"
>
  <rect
    height="1"
    rx="0"
    ry="0"
    width={400}
    x="20"
    y="175"
  />
  <rect
    height="100"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="40"
    y="75"
  />
  <rect
    height="50"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="80"
    y="125"
  />
  <rect
    height="70"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="120"
    y="105"
  />
  <rect
    height="140"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="160"
    y="35"
  />
  <rect
    height="120"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="200"
    y="55"
  />
  <rect
    height="160"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="240"
    y="15"
  />
  <rect
    height="40"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="280"
    y="135"
  />
  <rect
    height="90"
    rx="0"
    ry="0"
    width={30.76923076923077}
    x="320"
    y="85"
  />
</ContentLoader>
`;

exports[`Chart Loader tests renders correctly while props changed 2`] = `
<ContentLoader
  height={300}
  speed={1}
  viewBox="0 0 300 300"
  width="100%"
>
  <rect
    height="1"
    rx="0"
    ry="0"
    width={300}
    x="20"
    y="175"
  />
  <rect
    height="100"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="40"
    y="75"
  />
  <rect
    height="50"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="80"
    y="125"
  />
  <rect
    height="70"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="120"
    y="105"
  />
  <rect
    height="140"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="160"
    y="35"
  />
  <rect
    height="120"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="200"
    y="55"
  />
  <rect
    height="160"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="240"
    y="15"
  />
  <rect
    height="40"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="280"
    y="135"
  />
  <rect
    height="90"
    rx="0"
    ry="0"
    width={23.076923076923077}
    x="320"
    y="85"
  />
</ContentLoader>
`;
